apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl

```diff
@@ -279,34 +303,54 @@ on_kafka_ack(_Partition, buffer_overflow_discarded, _Callback) ->
     %% do not apply the callback (which is basically to bump success or fail counter)
     ok.
 
-on_get_status(_InstId, #{client_id := ClientId, kafka_topic := KafkaTopic}) ->
+on_get_status(_InstId, #{client_id := ClientId} = State) ->
     case wolff_client_sup:find_client(ClientId) of
         {ok, Pid} ->
-            do_get_status(Pid, KafkaTopic);
+            do_get_status(Pid, State);
         {error, _Reason} ->
             disconnected
     end.
 
-do_get_status(Client, KafkaTopic) ->
+do_get_status(Client, #{kafka_topic := KafkaTopic} = State) ->
+    CheckIfTopicExists =
+        fun(Pid, Topic, Default) ->
+            case wolff_client:check_if_topic_exists(Pid, Topic) of
```

Comment:
the return value match of wolff_client:check_if_topic_exists is repeated, maybe create a wrapper function instead.